Autopilot/PathManager/Src/CommWithAttitudeManager.cpp

```diff
@@ -32,27 +37,28 @@ void SendCommandsForAM(CommandsForAM *commands)
     *commandsOut = *commands;
 
     //Post mail slot to mail queue
-    osMailPut(commandsMailQ, commandsOut);
+    osStatus sendStatus = osMailPut(commandsMailQ, commandsOut);
 }
 
-bool GetAttitudeData(AttitudeData *data)
+
+bool GetFromPMToAM(CommandsForAM *commands)
 {
-    //Try to get data from mail queue
+    //Try to get commands from mail queue
     osEvent event;
-    AttitudeData * dataIn;
-    event = osMailGet(attitudeDataMailQ, 0);
+    CommandsForAM * commandsIn;
+    event = osMailGet(commandsMailQ, 0);
     if(event.status == osEventMail)
     {
-        dataIn = static_cast<AttitudeData *>(event.value.p);
+        commandsIn = static_cast<CommandsForAM *>(event.value.p);
         
-        //Keep the data and remove it from the queue
-        *data = *dataIn;
-        osMailFree(attitudeDataMailQ, dataIn);
+        //Keep the command and remove it from the queue
+        *commands = *commandsIn;
```

Comment:
I think something's wrong here. To me it looks like your'e reading and writing to/from the same queue, rather than actually exchange data with the other task.